packages/nextjs/pages/index.tsx

```diff
@@ -121,6 +130,15 @@ const Home: NextPage = () => {
     }
   };
 
+  const handleVerifiedContractInput = (input: string) => {
+    setVerifiedContractInput(input);
+    if (isAddress(input)) {
+      setVerifiedContractAddress(input);
+    } else {
+      setVerifiedContractAddress("");
+    }
+  };
+
```

Comment:
Do we really need this function and an extra state `verifiedContractInput` ? 

We are already doing the same check in `useEffect` before fetching abi : 

https://github.com/BuidlGuidl/abi.ninja/blob/7bcef6dd3c23bf623ed6ec471ef584f8eccf136d/packages/nextjs/pages/index.tsx#L65

So I think having only `verifiedContractAddress` should work 

or there was a bug because of which we added this?

Reply:
Yes, I had the same thought, but I was wondering if I might have missed anything.

Reply:
During my tests, I was having problems in this use case:

- Enter a valid `verifiedContractAddress`, then "Load Contract" button gets enabled.
- Change address to a invalid one, or delete it from the input.
- "Load Contract" button would remain enabled, and getting error when clicked because the input was not valid.

My guess was that the problem was because `useState` for `setVerifiedContractAddress` expects an `<Address>` type, so ´verifiedContractAddress´ was not getting updated. I might be totally wrong there 😅, but that's why I added this intermediate step.

Probably there is a much easier way to resolve it.

Reply:
Ohhhh yeah !! Thanks as always for giving the clear reproducible steps love it !!

We actually were not resetting `isAbiAvailable` if the entered address was changed from verified => incorrect address, just added an `else` condition in the same `useEffect` to reset `isAbiAvailable`  at https://github.com/BuidlGuidl/abi.ninja/pull/35/commits/120efc7ff68cbcceb06e40995e645672a8f30928

Reply:
Clean solution!!! Don't know how I could not sight it before adding all that unnecessary code! 😅 Thanks again ♥